src/main/java/jimmy/mcgymmy/logic/parser/ParserUtil.java

```diff
@@ -52,48 +52,65 @@ public static Name parseName(String name) throws ParseException {
     }
 
     /**
-     * Parses a {@code String phone} into a {@code Phone}.
-     * Leading and trailing whitespaces will be trimmed.
+     * Gets integer from string value.
      *
-     * @throws ParseException if the given {@code phone} is invalid.
+     * @param value        String containing value of nutrient.
+     * @param errorMessage String containing the error message.
+     * @return Integer value of string.
+     * @throws ParseException if the value is invalid.
      */
-    public static Phone parsePhone(String phone) throws ParseException {
-        requireNonNull(phone);
-        String trimmedPhone = phone.trim();
-        if (!Phone.isValidPhone(trimmedPhone)) {
-            throw new ParseException(Phone.MESSAGE_CONSTRAINTS);
+    private static int getNutrientValue(String value, String errorMessage) throws ParseException {
+        requireNonNull(value);
+        String trimmedValue = value.trim();
+        int nutrientValue;
+        try {
+            nutrientValue = Integer.parseInt(trimmedValue);
+        } catch (NumberFormatException numberFormatException) {
+            throw new ParseException(errorMessage);
         }
-        return new Phone(trimmedPhone);
+        return nutrientValue;
+    }
+
+    /**
+     * Parses a {@code String protein} into a {@code Protein}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code Protein} is invalid.
+     */
+    public static Protein parseProtein(String protein) throws ParseException {
+        int proteinValue = getNutrientValue(protein, Protein.MESSAGE_CONSTRAINTS);
+        return new Protein(proteinValue);
     }
 
+
     /**
-     * Parses a {@code String address} into an {@code Address}.
+     * Parses a {@code String carb} into an {@code Carbohydrate}.
      * Leading and trailing whitespaces will be trimmed.
      *
-     * @throws ParseException if the given {@code address} is invalid.
+     * @throws ParseException if the given {@code carb} is invalid.
      */
-    public static Address parseAddress(String address) throws ParseException {
-        requireNonNull(address);
-        String trimmedAddress = address.trim();
-        if (!Address.isValidAddress(trimmedAddress)) {
-            throw new ParseException(Address.MESSAGE_CONSTRAINTS);
+    public static Carbohydrate parseCarb(String carb) throws ParseException {
+        requireNonNull(carb);
+        String trimmedCarb = carb.trim();
```

Comment:
I think we should be using `getNutrientValue` here instead?

Reply:
What do you mean by this?